daemon/containerd/image_inspect.go

```diff
@@ -162,7 +165,16 @@ func (i *ImageService) collectRepoTagsAndDigests(ctx context.Context, tagged []c
 			log.G(ctx).WithError(err).Error("failed to create digested reference")
 			continue
 		}
-		repoDigests = append(repoDigests, reference.FamiliarString(digested))
+		repoDigestsMap[reference.FamiliarString(digested)] = struct{}{}
+	}
+
+	repoTags = make([]string, 0, len(repoTagsMap))
+	repoDigests = make([]string, 0, len(repoDigestsMap))
+	for tag := range repoTagsMap {
+		repoTags = append(repoTags, tag)
+	}
+	for digest := range repoDigestsMap {
+		repoDigests = append(repoDigests, digest)
 	}
```

Comment:
As we're using a map as intermediate, do we need to sort these afterwards to keep a stable result?

Reply:
Hmmm, right. In this case I think it's better to just go with the slice-based implementation: https://github.com/moby/moby/pull/48264#discussion_r1934218435

Reply:
Maybe? Are the values sorted when we receive them, or are they equally random? If we go for the slices, we could use the utility we have; https://github.com/moby/moby/blob/02563f2805a7d8a3078196d608e5e8fe08c71671/internal/sliceutil/sliceutil.go#L6-L16

Reply:
They're not really sorted AFAIK, but also not just random. I think it's good to preserve the order.

Reply:
Yep the slices approach also does use the `sliceutil.Dedup` at the end

Reply:
Possibly can use the `Dedup` utility that we implemented here; https://github.com/moby/moby/blob/02563f2805a7d8a3078196d608e5e8fe08c71671/internal/sliceutil/sliceutil.go#L6-L16